src/static-pages/index.tsx

```diff
@@ -0,0 +1,298 @@
+      }
+    }
+
+    if (path === `/${revisedType}`) {
+      const revisions = getRevisions(revisedConfig, lang, revisedType)
+      const current = revisions === null ? null : revisions[0]
+      const page = current === null ? null : await fetchContent(current)
+
+      if (page !== null) {
+        return new PreactResponse(RevisedPage(page))
+      } else {
+        return new NotFoundResponse()
+      }
+    }
+  }
+
+  return null
+}
+
+export function UnrevisedPage(page: WithContent<Page>) {
+  return (
+    <Template title={page.title} lang={page.lang}>
+      <h1>{page.title}</h1>
+      <div dangerouslySetInnerHTML={{ __html: page.content }} />
+    </Template>
+  )
+}
+
+export function RevisedPage(page: WithContent<RevisedPage>) {
+  return (
+    <Template title={page.title} lang={page.lang}>
+      <h1>
+        {page.title}{' '}
+        <small>
+          ({page.isCurrentRevision ? 'Current' : 'Archived'}
+          {' version of '}
+          {page.revision.toLocaleDateString(page.lang)})
+        </small>
+      </h1>
+      <div dangerouslySetInnerHTML={{ __html: page.content }} />
+    </Template>
+  )
+}
+
+export function RevisionsOverview(revisions: RevisedPage[]) {
+  const current = revisions[0]
+  const title = `Versions: ${current.title}`
+
+  return (
+    <Template title={title} lang={current.lang}>
+      <h1>{title}</h1>
+      There are the following archived versions of {current.title} available:
+      <ul>
+        {revisions.map(rev => {
+          const link = `/${rev.revisedType}/archiv/${getRevisionId(rev)}`
+
+          return (
+            <li>
+              <a href={link}>
+                {rev.revision.toLocaleDateString(rev.lang)}
+                {rev.isCurrentRevision ? ' (current version)' : ''}
+              </a>
+            </li>
+          )
+        })}
+      </ul>
+    </Template>
+  )
+}
+
+export async function fetchContent<A extends Page>(
+  page: A
+): Promise<WithContent<A> | null> {
+  const response = await fetch(new Request(page.url))
+
+  if (response.ok) {
+    const text = await response.text()
+    const rawContent = page.url.endsWith('.md') ? markdownToHtml(text) : text
+    const sanitizedContent = sanitizeHtml(rawContent)
+    const content = sanitizedContent.replace(
+      '__JS_GOOGLE_ANALYTICS_DEACTIVATE__',
+      'javascript:gaOptout();'
+    )
+
+    return { ...page, content }
+  } else {
+    return null
+  }
+}
+
+export function findRevisionById<A extends RevisedSpec>(
+  revisions: A[],
+  id: string
+): A | null {
+  return revisions.find(x => getRevisionId(x) === id) ?? null
+}
+
+export function getRevisionId(revised: RevisedSpec): string {
+  const year = revised.revision.getFullYear()
+  const month = revised.revision.getMonth() + 1
+  const day = revised.revision.getDate()
+
+  const paddedMonth = month.toString().padStart(2, '0')
+  const paddedDay = day.toString().padStart(2, '0')
+
+  return `${year}-${paddedMonth}-${paddedDay}`
+}
```

Comment:
Why do you need `revised.revision` to be a Date? Couldn't you just use the string `yyyy-mm-dd` as the revision?